src/app/shared/services/routes.service.ts

```diff
@@ -150,17 +145,22 @@ export class RoutesService implements ShowContentProvider {
   }
 
   private async updatePermissions(r: BffGeneratedRoute) {
```

Comment:
We have to talk about this function because this is not working how you have implemented it